### PR TITLE
fix: Passes day_of_week as required field to avoid 0 as default.

### PR DESCRIPTION
**Pull request by marcosuma**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s):

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: .changelog/2163.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+resource/maintenance_window: Fixes `day_of_week` param as **required** when calling the API
```
Comment (AgustinBettati) [inline .changelog/2163.txt]:
```suggestion
resource/mongodbatlas_maintenance_window: Fixes `day_of_week` param as **required** when calling the API
```

**Review by AgustinBettati — CHANGES_REQUESTED** [review]
LGTM, just small comment in changelog entry

File: internal/service/maintenancewindow/resource_maintenance_window_test.go [hunk 2]
```diff
@@ -40,15 +40,35 @@ func TestAccConfigRSMaintenanceWindow_basic(t *testing.T) {
...
 					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
-					resource.TestCheckResourceAttr(resourceName, "day_of_week", cast.ToString(dayOfWeekUpdated)),
+					resource.TestCheckResourceAttr(resourceName, "day_of_week", cast.ToString(dayOfWeek)),
 					resource.TestCheckResourceAttr(resourceName, "hour_of_day", cast.ToString(hourOfDayUpdated)),
 					resource.TestCheckResourceAttr(resourceName, "number_of_deferrals", "0"),
 				),
 			},
+			{
+				Config: configBasic(orgID, projectName, dayOfWeekUpdated, hourOfDay),
+				Check: resource.ComposeTestCheckFunc(
```
Comment (lantoli) [inline internal/service/maintenancewindow/resource_maintenance_window_test.go:54]: common check function, or helper functions like AddAttrChecks or AddAttrSetChecks can be used to reduce duplication, e.g.: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/doc_indentation/internal/service/searchindex/resource_search_index_test.go#L234

**Review by lantoli — APPROVED** [review]

**Review by AgustinBettati — APPROVED** [review]

File: internal/service/maintenancewindow/resource_maintenance_window_test.go (hunk 2 above)
Comment (marcosuma) [reply internal/service/maintenancewindow/resource_maintenance_window_test.go:54]: if it's ok I'd leave it like that - I like to keep tests siloed without having to jump to other functions to understand what's being asserted and how. Verbosity / code duplication in tests should not be an issue (tl;dr: write DAMP tests and DRY code)

Few references:
* https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html
* https://thomascannon.me/guides/verbose-tests

Comment (marcosuma) [reply internal/service/maintenancewindow/resource_maintenance_window_test.go:54]: But thanks for the comment!

Comment (lantoli) [reply internal/service/maintenancewindow/resource_maintenance_window_test.go:54]: you can define the checks at the beginning of the test and reuse in its steps.

**Review by JuliaMongo — APPROVED** [review]

**Pull request merged**